builds/release/detect-image-updates.yaml

```diff
@@ -0,0 +1,112 @@
+trigger: none
+pr: none
+
+schedules:
+# Twice daily (8AM/PM PDT)
+- cron: "0 3,15 * * *"
+  displayName: Twice daily check (8AM/PM PDT)
+  branches:
+    include:
+    - main
+    - release/*
+  always: true
+
+resources:
+  repositories:
+  - repository: product
+    type: github
+    endpoint: Azure/azure-iotedge
+    name: Azure/azure-iotedge
+
+variables:
+  NugetSecurityAnalysisWarningLevel: warn
+  DisableDockerDetector: true
+
+pool:
+  name: $(pool.linux.name)
+  demands:
+  - ImageOverride -equals agent-aziotedge-ubuntu-20.04-docker
+
+steps:
+- checkout: product
+  fetchDepth: 0
+
+- script: |
+    #!/bin/bash
+    set -euo pipefail
+
+    branch="${BUILD_SOURCEBRANCH#refs/heads/}"
+
+    # transform product-versions.json into a list of images for each product in this branch
+    images_json=$(cat product-versions.json | jq --arg registry "$REGISTRY" --arg branch "$branch" '
+      def is_image: has("type") and .type == "dockerImage";
+      def platforms: .componentTypes[] | select(.name == "dockerImage") | .platforms[] | { os, arch: .arch[] | . };
+      [
+        .channels[] | .products[] | select(
+          .components | any(is_image and .repo == "Azure/iotedge" and .branch == $branch)
+        ) | {
+          product: .id, version, images: [
+            platforms as $platforms | .components[] | select(is_image) | . as { $name, $version } | $platforms
+              | @text "\($registry)/\($name):\($version)-\(.os)-\(.arch)"
+          ] | sort
+        }
+      ] | unique
+    ')
+
+    # use build metadata present in each published image to determine if it is out of date
+    remove=( )
+    images=( $(echo "$images_json" | jq -r '[ .[].images ] | flatten | join("\n")') )
+    for image in ${images[@]}
+    do
+      echo "image: $image"
+
+      # This command assumes that the $image has a manifest, not a manifest list.
+      # In other words, $image is _not_ a multi-arch image.
+      read base_image current_digest <<< $(\
+        docker buildx imagetools inspect $image --format "{{json .BuildInfo}}" |
```

Comment:
So we get the base image and the current digest out of some json file? Then get the latest digest from the base image name? Adding more detailed comment as to the approach may help.

Reply:
The `docker buildx imagetools inspect` command directly examines an image (say, "mcr.microsoft.com/azureiotedge-agent:1.4.2-linux-amd64") and returns JSON (because of the `--format "{{json ...}}"` argument:

```json
{
  "frontend": "dockerfile.v0",
  "attrs": {
    "build-arg:EXE_DIR": ".",
    "filename": "Dockerfile",
    "source": "docker/dockerfile:1.4"
  },
  "sources": [
    {
      "type": "docker-image",
      "ref": "mcr.microsoft.com/dotnet/aspnet:6.0-alpine",
      "pin": "sha256:6032bb4619e4230855986f5ff67ea8eee345a476ab75b46c9977350cd503723a"
    }
  ]
}
```

From `.sources[0].ref` we know which base image Edge Agent targets, and from `.sources[0].pin` we know the digest of the specific image that was used when we built Edge Agent. We use jq to extract those two pieces of information into the variables `$base_image` and `$current_digest`.

We use the `docker buildx imagetools inspect` command again with `$base_image` to get the latest digest for that image. If it doesn't match `$current_digest`, that means the base image has changed since Edge Agent was last released.